atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SerializableTransactionManager.java

```diff
@@ -67,7 +67,11 @@
             this.initializationPrerequisite = initializationPrerequisite;
             this.callback = callBack;
             this.executorService = initializer;
-            scheduleInitializationCheckAndCallback();
+            if (isInitializedInternal()) {
```

Comment:
what is this doing? extract to a method?

as a side note: `scheduleInitializationCheckAndCallback` keeps scheduling itself, which is unnecessary since the executor service is a `ScheduledExecutorService`

Reply:
also this block exists elsewhere

Reply:
You need to give it a period if you want it repeated, which we could do but then we'd need to cancel it when it's successful so this looks nicer. Extracted duplicated code.